File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer;
    public int solution(int[] numbers, int target) {
        answer = 0;
        dfs(0, 0, numbers, target);
        return answer;
    }
    
    public static void dfs(int cnt, int sum, int[] nums, int target) {
        if(cnt == nums.length) {
            if(sum == target) {
                answer++;
            }
            return;
        }
        
        // 각 자리수마다
        dfs(cnt + 1, sum + nums[cnt], nums, target); // 더하거나
        dfs(cnt + 1, sum - nums[cnt], nums, target); // 빼거나
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    function dfs(cnt, sum) {
        if(cnt === numbers.length) {
            if(sum === target) {
                answer++;
            }
            return;
        }
        
        dfs(cnt + 1, sum + numbers[cnt]);
        dfs(cnt + 1, sum - numbers[cnt]);
    }
    
    dfs(0, 0);
    return answer;
}